fv3core/stencils/delnflux.py

```diff
@@ -1,218 +1,627 @@
-from gt4py.gtscript import PARALLEL, computation, interval
+from typing import Optional
+
+import gt4py.gtscript as gtscript
+import numpy as np
+from gt4py.gtscript import (
+    __INLINED,
+    FORWARD,
+    PARALLEL,
+    computation,
+    horizontal,
+    interval,
+    region,
+)
 
 import fv3core._config as spec
 import fv3core.utils.corners as corners
 import fv3core.utils.gt4py_utils as utils
-from fv3core.decorators import gtstencil
-from fv3core.stencils.basic_operations import copy_stencil
-from fv3core.utils.typing import FloatField, FloatFieldIJ
-
-
-@gtstencil
-def fx2_order(q: FloatField, del6_v: FloatFieldIJ, fx2: FloatField, order: int):
-    with computation(PARALLEL), interval(...):
-        fx2[0, 0, 0] = del6_v * (q[-1, 0, 0] - q)
-        fx2[0, 0, 0] = -1.0 * fx2 if order > 1 else fx2
+from fv3core.decorators import FrozenStencil, gtstencil
+from fv3core.utils.grid import axis_offsets
+from fv3core.utils.typing import FloatField, FloatFieldIJ, FloatFieldK
+
+
+def calc_damp(damp4: FloatField, nord: FloatFieldK, damp_c: FloatFieldK, da_min: float):
+    with computation(FORWARD), interval(...):
+        damp4 = (damp_c * da_min) ** (nord + 1)
+
+
+def fx_calc_stencil_region(q: FloatField, del6_v: FloatFieldIJ, fx: FloatField):
```

Comment:
is `fx_calc_stencil_region` the right name? both stencil and region are somehow details right?